### PR TITLE
Use separate proxy settings for Maven repository when enabled

### PR DESCRIPTION
**Pull request by mdonkers**
Replacing PR https://github.com/instana/instana-agent-docker/pull/14 to correctly introduce and use the Maven proxy variables.

Need to be enabled explicitly, as otherwise it will use the 'default' Agent proxy settings.

### CODE REVIEW

**Comment by mdonkers** [comment]
One additional note on the `INSTANA_REPOSITORY_PROXY_ENABLED` flag:

What especially should be clear (also from the documentation) is that the `INSTANA_REPOSITORY_PROXY_ENABLED` flag must be set if the (Maven) repository needs a _different_ configuration than the Agent proxy. When they can be the same, nothing needs to be configured for the `REPOSITORY_PROXY`.
On one hand this might be confusing, but as before we did not have separate configuration, existing customers just updating their Docker image would at least get similar behaviour (instead of having to configure a second proxy).

File: rhel/run.sh [hunk 1]
```diff
@@ -45,6 +45,21 @@ if [ -z "${INSTANA_DOWNLOAD_KEY}" ]; then
   INSTANA_DOWNLOAD_KEY="${INSTANA_AGENT_KEY}"
 fi
 
+# Take over Agent Proxy variables if no Repository Proxy is enabled
+case ${INSTANA_REPOSITORY_PROXY_ENABLED} in
+  y|Y|yes|Yes|YES|1|true)
```
Comment (dlbock) [inline rhel/run.sh]: @mdonkers what about `enabled` ? :trollface:

Comment (mdonkers) [reply rhel/run.sh]: `ENABLED=enabled` ?? Nah! It's no JavaScript :-P

Comment (dlbock) [reply rhel/run.sh]: 🤣

**Pull request merged**